tests/components/bond/test_light.py

```diff
@@ -84,10 +92,54 @@ async def test_update_reports_light_is_off(hass: core.HomeAssistant):
     """Tests that update command sets correct state when Bond API reports the light is off."""
     await setup_platform(hass, LIGHT_DOMAIN, ceiling_fan("name-1"))
 
-    with patch(
-        "homeassistant.components.bond.Bond.getDeviceState", return_value={"light": 0}
-    ):
+    with patch_bond_device_state(return_value={"light": 0}):
         async_fire_time_changed(hass, utcnow() + timedelta(seconds=30))
         await hass.async_block_till_done()
 
     assert hass.states.get("light.name_1").state == "off"
+
+
+async def test_turn_on_fireplace(hass: core.HomeAssistant):
+    """Tests that turn on command delegates to API."""
+    await setup_platform(
+        hass, LIGHT_DOMAIN, fireplace("name-1"), bond_device_id="test-device-id"
+    )
+
+    with patch_bond_turn_on() as mock_turn_on, patch_bond_set_flame() as mock_set_flame, patch_bond_device_state():
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "light.name_1", ATTR_BRIGHTNESS: 128},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+
+    mock_turn_on.assert_called_once()
+    mock_set_flame.assert_called_once_with("test-device-id", 50)
+
+
+async def test_turn_off_fireplace(hass: core.HomeAssistant):
+    """Tests that turn off command delegates to API."""
+    await setup_platform(hass, LIGHT_DOMAIN, fireplace("name-1"))
+
+    with patch_bond_turn_off() as mock_turn_off, patch_bond_device_state():
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: "light.name_1"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        mock_turn_off.assert_called_once()
+
+
+async def test_flame_converted_to_brightness(hass: core.HomeAssistant):
+    """Tests that reported flame level (0..100) converted to HA brightness (0...255)."""
+    await setup_platform(hass, LIGHT_DOMAIN, fireplace("name-1"))
+
+    with patch_bond_device_state(return_value={"power": 1, "flame": 50}):
+        async_fire_time_changed(hass, utcnow() + timedelta(seconds=30))
+        await hass.async_block_till_done()
+
+    _LOGGER.warning(hass.states.get("light.name_1").attributes)
+    assert hass.states.get("light.name_1").attributes[ATTR_BRIGHTNESS] == 128
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)